### PR TITLE
Document styling changes in 4.0

### LINKED ISSUE #11681: [CPQ] [4.0] cleanUp deprecated code
With the CPQ release (3.3) we deprecated some code, especially constructors, to inject additional dependencies.
with 4.0 we should revisit this.

Clean up is done in related update schematics issues (#11232, #12198, #12199)

**Styling changes:**
- [ ] Add styling changes from this [Wiki page](https://cxwiki.sap.com/pages/viewpage.action?spaceKey=prodandtech&title=Changes+to+styles+in+4.0) into `/docs/migration/css-changes-in-version-4.md` file

### PR DESCRIPTION
**Pull request by Larisa-Staroverova**

### CODE REVIEW

File: docs/migration/css-changes-in-version-4.md [hunk 1]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
 ## Changes in Checkout Components
 
 * `cx-product-variants` selector has been moved to corresponding feature-lib `@spartacus/product`.
+
+## Change in Configurator Attribute Type Components
+
+* `cx-quantity` selector has been added to align its padding to the corresponding attribute type.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: Please re-phrase, don't know what's exactly meant

File: docs/migration/css-changes-in-version-4.md [hunk 2]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
...
+
+## Changes in Configurator Product Title Component
+
+* `width` set to 80% on `%cx-configurator-product-title` to use only 80% of the configuration product title width.
+
+* `button` instead of the anchor link on `%cx-configurator-product-title`.
+
+* `padding` set to 16px/ 32px on `%cx-configurator-product-title` for `cx-details` selector to align spacing depending on the screen size.
+
+* `cx-configurator-image` mixin has been defined on `%cx-configurator-product-title` for `cx-details-image` selector to define the unique styling.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: Please explain what you mean with 'the unique styling'

File: docs/migration/css-changes-in-version-4.md [hunk 3]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
...
+
+* `padding` set to 0px and `margin` to 17px on `%cx-configurator-attribute-type` to align the spacing between the configuration attribute types.
+
+## Change in Configurator Attribute Drop-Down Component
+
+* `padding` set to 1rem on `%cx-configurator-attribute-drop-down` for `cx-configurator-attribute-quantity` selector to define the spacing between the drop-down attribute type, and the quantity counter.
+
+## Change in Configurator Attribute Checkbox List Component
+
+* `padding` set to 1rem on `%cx-configurator-attribute-checkbox-list` to define the spacing between the checkbox-list attribute type, and the quantity counter.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: remove last comma

File: docs/migration/css-changes-in-version-4.md [hunk 4]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
...
+
+## Change in Configurator Attribute Radio Button Component
+
+* `padding` set to 1rem on `%cx-configurator-attribute-radio-button` to define the spacing between the radio-button attribute type, and the quantity counter.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: remove last comma

File: docs/migration/css-changes-in-version-4.md [hunk 5]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
...
+
+* `padding` set to 16px on `%cx-configurator-footer-container` mixin to align the spacing between the price summary and add-to-cart button.
+
+## Change in Configurator Required Error Message Mixin
+
+* `padding` set to 5px on `%cx-configurator-required-error-msg` mixin to add the spacing at the end of the cx-icon selector.
+
+## Changes in Configurator Overview Form Component
+
+* `padding` set to 0px on `%cx-configurator-overview-form` to prevent adding any spacing to the configuration overview form.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: What does that mean exactly?

File: docs/migration/css-changes-in-version-4.md [hunk 6]
```diff
@@ -5,3 +5,85 @@ title: Changes to Styles in 4.0
...
+
+* `padding` set to 32px/16px on `%cx-configurator-overview-form` for `h2` selector to align spacing around the configuration overview group titles.
+
+* `cx-configurator-truncate-content` mixin has been added on `%cx-configurator-overview-form` for `span` selector to enable the overview group title truncation for the small widgets.
+
+* `padding` set to 32px on `%cx-configurator-overview-form` for `cx-attribute-value-pair` selector to align spacing between the configuration overview attribute value pairs.
+
+* `display` set to `none / inline` and `visibility` to `hidden` on `%cx-configurator-overview-form` for `cx-attribute-value-pair` selector to define the visibility for the configuration overview attribute value label.
+
+* `padding` set to 20px on `%cx-configurator-overview-form` for `cx-no-attribute-value-pair` selector to align spacing between the configuration overview no attribute value pairs.
```
Comment (ChristophHi) [inline docs/migration/css-changes-in-version-4.md]: 'no attribute value pairs'?

**Review by ChristophHi — CHANGES_REQUESTED** [review]

File: docs/migration/css-changes-in-version-4.md (hunk 3 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: If I remove a comma the WebStorm is quite unhappy about it and shows the following remark:
image

File: docs/migration/css-changes-in-version-4.md (hunk 4 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: If I remove a comma the WebStorm is quite unhappy (see a WebStorm remark above)

File: docs/migration/css-changes-in-version-4.md (hunk 1 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Do you have a better idea? I don't :(

File: docs/migration/css-changes-in-version-4.md (hunk 2 above)
Comment (ChristophHi) [reply docs/migration/css-changes-in-version-4.md]: 'to achieve a consistent styling'

File: docs/migration/css-changes-in-version-4.md (hunk 5 above)
Comment (ChristophHi) [reply docs/migration/css-changes-in-version-4.md]: 'to fix inconsistent spacings in the configuration overview form'

File: docs/migration/css-changes-in-version-4.md (hunk 3 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

File: docs/migration/css-changes-in-version-4.md (hunk 2 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

File: docs/migration/css-changes-in-version-4.md (hunk 4 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

File: docs/migration/css-changes-in-version-4.md (hunk 5 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

File: docs/migration/css-changes-in-version-4.md (hunk 1 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

File: docs/migration/css-changes-in-version-4.md (hunk 6 above)
Comment (Larisa-Staroverova) [reply docs/migration/css-changes-in-version-4.md]: Done!

**Review by ChristophHi — APPROVED** [review]

**Pull request merged**